File: modules/cubex-i18n/src/main/java/org/cubexmc/i18n/BookLiteLanguageAdapter.java
package org.cubexmc.i18n;

import java.util.List;
import java.util.Map;
import org.bukkit.command.CommandSender;

public class BookLiteLanguageAdapter {

    private final I18nService i18n;

    public BookLiteLanguageAdapter(I18nService i18n) {
        this.i18n = i18n;
    }

    public void setLocale(String locale) {
        i18n.setCurrentLocale(locale);
    }

    public void load() {
        i18n.reload();
    }

    public String raw(String key) {
        return i18n.raw(key);
    }

    public List<String> rawList(String key) {
        return i18n.rawList(key);
    }

    public String msg(String key) {
        return i18n.message(key);
    }

    public String msg(String key, Map<String, String> placeholders) {
        return i18n.message(key, placeholders);
    }

    public List<String> msgList(String key, Map<String, String> placeholders) {
        return i18n.messageList(key, placeholders);
    }

    public void send(CommandSender to, String key) {
        send(to, key, null);
    }

    public void send(CommandSender to, String key, Map<String, String> placeholders) {
        if (to != null) {
            to.sendMessage(msg(key, placeholders));
        }
    }
}

File: modules/cubex-i18n/src/main/kotlin/org/cubexmc/i18n/BookLiteLanguageAdapter.kt
package org.cubexmc.i18n

import org.bukkit.command.CommandSender

open class BookLiteLanguageAdapter(private val i18n: I18nService) {
    open fun setLocale(locale: String) = i18n.setCurrentLocale(locale)
    open fun load() = i18n.reload()
    open fun raw(key: String?): String = i18n.raw(key)
    open fun rawList(key: String?): List<String> = i18n.rawList(key)
    open fun msg(key: String?): String = i18n.message(key)
    open fun msg(key: String?, placeholders: Map<String, String>?): String = i18n.message(key, placeholders)
    open fun msgList(key: String?, placeholders: Map<String, String>?): List<String> =
        i18n.messageList(key, placeholders)
    open fun send(to: CommandSender?, key: String?) = send(to, key, null)
    open fun send(to: CommandSender?, key: String?, placeholders: Map<String, String>?) {
        if (to != null) to.sendMessage(msg(key, placeholders))
    }
}
